proxy.go

```diff
@@ -201,6 +201,8 @@ func (rp *reverseProxy) proxyRequest(s *scope, rw http.ResponseWriter, srw *stat
 			log.Errorf("%s: cannot kill query: %s; query: %q", s, err, q)
 		}
 		srw.statusCode = 499 // See https://httpstatuses.com/499 .
+		err = fmt.Errorf("client closed connection for query: %q", q)
```

Comment:
I believe the lines 204-205 are unnecessary, since `killQuery` on the line 200 will set `s.canceled`, so the cached response must be rolled back. Additionally, there is no sense in returning any response to the client, since the connection is closed.

The `proxyRequest` becomes too complicated, so it would be great to simplify it in further PR. For instance, think about removing `srw` argument from the function and using only `rw` for obtaining/setting status codes.